src/structures/Guild.js

```diff
@@ -936,12 +937,17 @@ class Guild {
     if (data.color) data.color = Util.resolveColor(data.color);
     if (data.permissions) data.permissions = Permissions.resolve(data.permissions);
 
-    return this.client.api.guilds(this.id).roles.post({ data, reason }).then(role =>
-      this.client.actions.GuildRoleCreate.handle({
+    return this.client.api.guilds(this.id).roles.post({ data, reason }).then(r => {
+      const { role } = this.client.actions.GuildRoleCreate.handle({
         guild_id: this.id,
-        role,
-      }).role
-    );
+        role: r,
+      });
+      if (data.position) {
+        role.setPosition(data.position, reason);
```

Comment:
You're right, i fixed it so that it returns the role#setPosition now. should be fine in terms of asynchronous operations.